x/stakeibc/keeper/hooks_deposit_records_test.go

```diff
@@ -0,0 +1,527 @@
+			DepositEpochNumber: currentEpoch,
+		},
+		{
+			Id:                 8,
+			Amount:             8000,
+			Denom:              Atom,
+			HostZoneId:         HostChainId,
+			Status:             recordstypes.DepositRecord_STAKE,
+			DepositEpochNumber: currentEpoch,
+		},
+	}
+
+	return TestDepositRecords{
+		emptyRecords:          emptyDepositRecords,
+		recordsToBeTransfered: recordsToBeTransfered,
+		recordsToBeStaked:     recordsToBeStaked,
+		recordsInCurrentEpoch: recordsInCurrentEpoch,
+		transferAmount:        transferAmount,
+		stakeAmount:           stakeAmount,
+	}
+}
+
+func (s *KeeperTestSuite) SetupDepositRecords() DepositRecordsTestCase {
+	delegationAccountOwner := fmt.Sprintf("%s.%s", HostChainId, "DELEGATION")
+	delegationChannelID := s.CreateICAChannel(delegationAccountOwner)
+	delegationAddress := s.IcaAddresses[delegationAccountOwner]
+
+	ibcDenomTrace := s.GetIBCDenom(Atom) // we need a true IBC denom here
+	hostModuleAddress := stakeibctypes.NewZoneAddress(HostChainId)
+	s.App.TransferKeeper.SetDenomTrace(s.Ctx(), ibcDenomTrace)
+
+	initialModuleAccountBalance := sdk.NewCoin(ibcDenomTrace.IBCDenom(), sdk.NewInt(15_000))
+	s.FundAccount(hostModuleAddress, initialModuleAccountBalance)
+
+	validators := []*stakeibctypes.Validator{
+		{
+			Name:    "val1",
+			Address: "gaia_VAL1",
+			Weight:  1,
+		},
+		{
+			Name:    "val2",
+			Address: "gaia_VAL2",
+			Weight:  4,
+		},
+	}
+
+	hostZone := stakeibctypes.HostZone{
+		ChainId:           HostChainId,
+		Address:           hostModuleAddress.String(),
+		DelegationAccount: &stakeibctypes.ICAAccount{Address: delegationAddress},
+		ConnectionId:      ibctesting.FirstConnectionID,
+		TransferChannelId: ibctesting.FirstChannelID,
+		HostDenom:         Atom,
+		IBCDenom:          ibcDenomTrace.IBCDenom(),
+		Validators:        validators,
+	}
+
+	currentEpoch := uint64(2)
+	strideEpochTracker := stakeibctypes.EpochTracker{
+		EpochIdentifier:    epochtypes.STRIDE_EPOCH,
+		EpochNumber:        currentEpoch,
+		NextEpochStartTime: uint64(s.Coordinator.CurrentTime.UnixNano() + 30_000_000_000), // dictates timeouts
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+	s.App.StakeibcKeeper.SetEpochTracker(s.Ctx(), strideEpochTracker)
+
+	initialDepositRecords := s.GetInitialDepositRecords(currentEpoch)
+	for _, depositRecord := range initialDepositRecords.GetAllRecords() {
+		s.App.RecordsKeeper.AppendDepositRecord(s.Ctx(), depositRecord)
+	}
+
+	return DepositRecordsTestCase{
+		initialDepositRecords:       initialDepositRecords,
+		initialModuleAccountBalance: initialModuleAccountBalance,
+		hostZone:                    hostZone,
+		hostModuleAddress:           hostModuleAddress,
+		epochNumber:                 currentEpoch,
+		TransferChannel: Channel{
+			PortID:    ibctesting.TransferPort,
+			ChannelID: ibctesting.FirstChannelID,
+		},
+		DelegationChannel: Channel{
+			PortID:    "icacontroller-" + delegationAccountOwner,
```

Comment:
nit: this is defined in the ICA module as `KeyPrefix` I believe

Reply:
Nice, I was wondering where to find it! Updated